src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -836,6 +836,11 @@ export abstract class _MatAutocompleteTriggerBase
         event.preventDefault();
       }
     });
+
+    // Subscribe to the pointer events stream so that it doesn't get picked up by other overlays.
+    // TODO(crisbeto): we should switch `_getOutsideClickStream` eventually to use this stream,
+    // but the behvior isn't exactly the same and it ends up breaking some internal tests.
+    overlayRef.outsidePointerEvents().subscribe();
```

Comment:
The outside click dispatcher keeps a stack of the open overlays. Whenever a click event happens, it starts from the top of the stack and works its way down until it finds an overlay that has any subscribers to `outsidePointerEvents` so that it can dispatch the event to it. This was a problem for the autocomplete, because it handles outside clicks internally, instead of going through the dispatcher which meant that it would get skipped over when determining which overlay is on top. We proper fix would be to use the dispatcher, but last time I tried it was somewhat breaking.

Reply:
Gotcha thanks